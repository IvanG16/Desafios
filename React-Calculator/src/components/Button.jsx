import React from 'react';
import PropTypes from 'prop-types';

export function Button({ value, handleClick, valueType }) {
  const onClick = () => {
    handleClick(value);
  };

  const styles = `w-[81px] h-[45px] p-2.5 rounded-[5px] border border-black flex-col justify-center items-center gap-2.5 inline-flex ${valueType === 'operand' ? 'bg-stone-50' : 'bg-stone-300'} `;

  return (
    <button onClick={onClick} type="button" className={styles}>
      <div className="w-1/2 text-black text-2xl font-normal Inconsolata">
        {value}
      </div>
    </button>
  );
}

Button.propTypes = {
  handleClick: PropTypes.func.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  valueType: PropTypes.oneOf(['number', 'operand']).isRequired,
};

export default Button;
