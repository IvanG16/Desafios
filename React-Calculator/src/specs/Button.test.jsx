import Button from '../components/Button'
import { describe, it, expect } from 'vitest'
import { render } from '@testing-library/react'

/**
 * @vitest-environment jsdom
 */

describe('<Button />', ()=>{
    it('renders properly',()=> {
        const { container } = render(<Button />)
        expect(container).toMatchSnapshot()
    })
})
