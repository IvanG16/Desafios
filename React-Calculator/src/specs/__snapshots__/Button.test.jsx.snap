// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Button /> > renders properly 1`] = `
<div>
  <button
    class="w-[81px] h-[45px] p-2.5 rounded-[5px] border border-black flex-col justify-center items-center gap-2.5 inline-flex bg-stone-300 "
    type="button"
  >
    <div
      class="w-1/2 text-black text-2xl font-normal Inconsolata"
    />
  </button>
</div>
`;
